common/djangoapps/third_party_auth/middleware.py

```diff
@@ -23,3 +31,19 @@ def get_redirect_uri(self, request, exception):
             redirect_uri = pipeline.AUTH_DISPATCH_URLS[auth_entry]
 
         return redirect_uri
+
+    def process_exception(self, request, exception):
+        """Handles specific exception raised by Python Social Auth eg HTTPError."""
+
+        referer_url = request.META.get('HTTP_REFERER', '')
+        if referer_url and isinstance(exception, HTTPError) and \
```

Comment:
Please use parenthesis rather than \

```
if (referer_url and isinstance(exception, HTTPError) and 
    exception.response.status_code == 502):
```